### ISSUE TITLE
E1136 false positive on PySide6 overloaded signal subscript

### ISSUE THREAD

**Issue by heapwalker20** [pylint-dev/pylint#10916]
### Bug description
Hello and thank you for the project! I have the following minimal reporduction code- which leads to some false postives:
```python
from PySide6.QtWidgets import QApplication, QWidget, QVBoxLayout, QSpinBox, QComboBox, QLabel

class SettingsPanel(QWidget):
    def __init__(self):
        super().__init__()
        layout = QVBoxLayout(self)

        self.font_size = QSpinBox()
        self.font_size.setRange(8, 72)
        self.font_size.setValue(12)
        layout.addWidget(self.font_size)

        self.theme = QComboBox()
        self.theme.addItems(["Light", "Dark", "System"])
        layout.addWidget(self.theme)

        self.status = QLabel("Ready")
        layout.addWidget(self.status)

        # these trigger E1136
        self.font_size.valueChanged[int].connect(self.on_font_changed)
        self.theme.currentIndexChanged[int].connect(self.on_theme_changed)

    def on_font_changed(self, size):
        self.status.setText(f"Font size: {size}pt")

    def on_theme_changed(self, index):
        self.status.setText(f"Theme: {self.theme.itemText(index)}")

if __name__ == "__main__":
    app = QApplication([])
    panel = SettingsPanel()
    panel.show()
    app.exec()
```

Please advice how to resolve.

### Configuration

```ini

```

### Command used

```shell
pylint --extension-pkg-allow-list=PySide6 settings_panel.py
```

### Pylint output

```python
settings_panel.py:21:8: E1136: Value 'self.font_size.valueChanged' is unsubscriptable (unsubscriptable-object)
settings_panel.py:22:8: E1136: Value 'self.theme.currentIndexChanged' is unsubscriptable (unsubscriptable-object)
```

### Expected behavior

No error. The subscript syntax should be correct.

### Pylint version

```shell
pylint 3.3.6
astroid 3.3.9
Python 3.12.0
```

### Additional dependencies

```python
pyside==6.10.2
```

**Comment by emmanuel-ferdman** [comment]
@heapwalker20 `pylint-qt` plugin was just released and is available on [PyPI](https://pypi.org/project/pylint-qt/)! 🎉
It fixes those Qt false-positives. Please give it a try:
```sh
pip install pylint-qt
pylint --load-plugins=pylint_qt --extension-pkg-allow-list=PySide6 your_code.py
```
Or in `pyproject.toml`:
```
[tool.pylint.main]
load-plugins = ["pylint_qt"]
extension-pkg-allow-list = ["PySide6"]
```
repo: https://github.com/pylint-dev/pylint-qt
